src/main/kotlin/nl/info/client/brp/BrpClientService.kt

```diff
@@ -104,18 +106,24 @@ class BrpClientService @Inject constructor(
         auditEvent: String,
         defaultPurpose: String?,
         extractPurpose: (ZaakafhandelParameters) -> String?
-    ): String? = Regex("""ZAAK-\d{4}-\d+""")
-        .find(auditEvent)
-        ?.value
-        ?.let { zaakIdentificatie ->
-            runCatching {
-                val zaak = zrcClientService.readZaakByID(zaakIdentificatie)
-                val parameters = zaakafhandelParameterService.readZaakafhandelParameters(
-                    zaak.zaaktype.extractUuid()
-                )
-                extractPurpose(parameters)
-            }.getOrNull()
-        } ?: defaultPurpose
+    ): String? =
+        auditEvent
+            .also { LOG.info("Resolving purpose for audit event: $it") }
+            .let { """ZAAK-\d{4}-\d+""".toRegex().find(it)?.value }
+            ?.runCatching {
+                zrcClientService.readZaakByID(this)
+                    .zaaktype.extractUuid()
+                    .let(zaakafhandelParameterService::readZaakafhandelParameters)
+                    .let(extractPurpose)
+            }?.onFailure {
+                LOG.warning("Failed to resolve purpose from audit event '$auditEvent': ${it.message}")
```

Comment:
Log an error on a including the cause.

(I know you copied it from my suggestion, sorry I didn't put that in there immediately)

Reply:
Decided to use a warning instead, as it will not stop the process. 
We do need to capture the stack trace though, so that will be included.